src/Security/RolesBuilder/SecurityRolesBuilder.php

```diff
@@ -54,48 +54,69 @@ public function __construct(
         $this->rolesHierarchy = $rolesHierarchy;
     }
 
-    public function getPermissionLabels(): array
+    public function getExpandedRoles(?string $domain = null): array
     {
-        return [];
-    }
-
-    public function getRoles(string $domain = null, bool $expanded = true): array
-    {
-        $baseRoles = [$this->pool->getOption('role_super_admin') => [],
-            $this->pool->getOption('role_admin') => [], ];
-        $hierarchy = array_merge($baseRoles, $this->rolesHierarchy);
-
         $securityRoles = [];
-        foreach ($hierarchy as $role => $childRoles) {
+        foreach ($hierarchy = $this->getHierarchy() as $role => $childRoles) {
+            $translatedRoles = array_map(
+                [$this, 'translateRole'],
+                $childRoles,
+                array_fill(0, count($childRoles), $domain)
+            );
+
             $securityRoles[$role] = [
                 'role' => $role,
-                'role_translated' => $this->translateRole($role, $domain),
+                'role_translated' => $role.': '.implode(', ', $translatedRoles),
```

Comment:
Kept the same trailing `: `, not sure what to do…

Reply:
check before adding `:`?

```
            $translatedRoles = count($translatedRoles) > 0 ? ': '.implode(', ', $translatedRoles) : '';
             $securityRoles[$role] = [
                 'role' => $role,
                 'role_translated' => $role. $translatedRoles
                 'is_granted' => $this->authorizationChecker->isGranted($role),
             ];
```